pkg/health/condition/check_backup_ready.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package condition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type backupReadyCheck struct {
+	cl client.Client
+}
+
+const (
+	FullBackupSucceeded string = "FullBackupSucceeded"
+	IncrBackupSucceeded string = "IncrementalBackupSucceeded"
+	BackupFailed        string = "BackupFailed"
+	Unknown             string = "Unknown"
+)
+
+func (a *backupReadyCheck) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
+	//Default case
+	result := &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  druidv1alpha1.ConditionUnknown,
+		reason:  Unknown,
+		message: "Cannot determine etcd backup status since no snapshot leases present",
+	}
+
+	//Fetch snpashot leases
+	fullSnapLease := &coordinationv1.Lease{}
+	deltaSnapLease := &coordinationv1.Lease{}
+	var fullSnapErr, incrSnapErr error
+	fullSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getFullSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, fullSnapLease)
+	incrSnapErr = a.cl.Get(ctx, types.NamespacedName{Name: getDeltaSnapLeaseName(&etcd), Namespace: etcd.ObjectMeta.Namespace}, deltaSnapLease)
+
+	if fullSnapErr != nil || incrSnapErr != nil || (fullSnapLease.Spec.HolderIdentity == nil && deltaSnapLease.Spec.HolderIdentity == nil) {
```

Comment:
I wonder about how this behaves for new clusters as I expect there will be a period where only one of the leases (usually `fullSnapLease.Spec.HolderIdentity`) is maintained. Then we'd add a failing backup condition until the next snapshot happens, right? Should we be more tolerant in this case?

Reply:
Hm, yes, you're right there
Will rework the conditions that follow to be a bit more tolerant